src/main/java/com/zendesk/maxwell/SynchronousBootstrapper.java

```diff
@@ -0,0 +1,200 @@
+		return context.getConnectionPool().getConnection();
+	}
+
+	private RowMap replicationStreamBootstrapStartRow(Table table, BinlogPosition position) {
+		return replicationStreamBootstrapRow("bootstrap-start", table, position);
+	}
+
+	private RowMap replicationStreamBootstrapCompletedRow(Table table, BinlogPosition position) {
+		return replicationStreamBootstrapRow("bootstrap-complete", table, position);
+	}
+
+	private RowMap replicationStreamBootstrapRow(String type, Table table, BinlogPosition position) {
+		return new RowMap(
+				type,
+				table.getDatabase().getName(),
+				table.getName(),
+				System.currentTimeMillis(),
+				table.getPKList(),
+				position);
+	}
+
+	@Override
+	public void completeBootstrap(RowMap completeBootstrapRow, Schema schema, AbstractProducer producer, OpenReplicator replicator) throws Exception {
+		String databaseName = ( String ) completeBootstrapRow.getData("database_name");
+		String tableName = ( String ) completeBootstrapRow.getData("table_name");
+		Database database = findDatabase(schema, databaseName);
+		ensureTable(tableName, findDatabase(schema, databaseName));
+		Table table = findTable(tableName, database);
+		BinlogPosition position = new BinlogPosition(replicator.getBinlogPosition(), replicator.getBinlogFileName());
+		producer.push(completeBootstrapRow);
+		producer.push(replicationStreamBootstrapCompletedRow(table, position));
+		LOGGER.info(String.format("bootstrapping ended for %s.%s", databaseName, tableName));
+	}
+
+	@Override
+	public void resume(Schema schema, AbstractProducer producer, OpenReplicator replicator) throws Exception {
+		try ( Connection connection = context.getConnectionPool().getConnection() ) {
+			// This update resets all rows of incomplete bootstraps to their original state.
+			// These updates are treated as fresh bootstrap requests and trigger a restart
+			// of the bootstrap process from the beginning.
+			String sql = "update maxwell.bootstrap set started_at = NULL where is_complete = 0 and started_at is not NULL";
+			connection.prepareStatement(sql).execute();
+		}
+	}
+
+	private Table findTable(String tableName, Database database) {
+		Table table = database.findTable(tableName);
+		if ( table == null )
+			throw new RuntimeException("Couldn't find table " + tableName);
+		return table;
+	}
+
+	private Database findDatabase(Schema schema, String databaseName) {
+		Database database = schema.findDatabase(databaseName);
+		if ( database == null )
+			throw new RuntimeException("Couldn't find database " + databaseName);
+		return database;
+	}
+
+	private void ensureTable(String tableName, Database database) {
+		findTable(tableName, database);
+	}
+
+	private ResultSet getAllRows(String databaseName, String tableName, Connection connection) throws SQLException, InterruptedException {
+		Statement statement = createBatchStatement(connection);
+		return statement.executeQuery(String.format("select * from %s.%s", databaseName, tableName));
```

Comment:
let's go ahead and order by the table's primary key here.  If you leave off any ordering it's just "database order" which is largely undefined, although often it can be ordered by the PK.